Mathlib/Util/AliasIn.lean

```diff
@@ -0,0 +1,55 @@
+/-
+Copyright (c) 2025 Floris van Doorn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn
+-/
+module
+
+public meta import Mathlib.Lean.Expr.Basic
+
+/-! ## The `@[alias_in]` attribute -/
+
+public meta section
+
+/-- Adds an alias of this declaration in a different namespace.
+Example:
+```
+@[alias_in Foo.Bar] def A.B.C.d := ...
+```
+behaves like
+```
+alias A.Foo.Bar.d := A.B.C.d
+```
+We replace the rightmost/innermost namespaces, but always leave the final part of the name intact.
+By default, `alias_in` assumes that we want to replace the same number of namespaces from the
+original name as given in the new namespace. You can override this by adding a number at the end
+```
+@[alias_in Foo.Bar 3] def A.B.C.d := ...
+```
+behaves like
+```
+alias Foo.Bar.d := A.B.C.d
+```
+-/
+syntax (name := aliasIn) "alias_in " ident (ppSpace num)? : attr
+
+open Lean Meta Elab Command
+@[inherit_doc aliasIn]
+initialize registerBuiltinAttribute {
+    name := `aliasIn
+    descr := "create alias in another namespace"
+    applicationTime := .afterCompilation
+    add := fun
+    | src, `(attr| alias_in%$tk $nm $[$num]?), _ => do
```

Comment:
```suggestion
    | src, `(attr| alias_in%$tk $nm $[$num]?), k => do
      unless k == .global do
        throwError m!"The `alias_in` attribute cannot be {k}."
```
what do you think of throwing an explicit error when `scoped` or `local` is added?

If so, then you could also add the tests
```lean
/-- error: The `alias_in` attribute cannot be local. -/
#guard_msgs in
@[local alias_in Baz] def nonlocal : Nat := 2

/-- error: The `alias_in` attribute cannot be scoped. -/
#guard_msgs in
@[scoped alias_in Baz] def non.scoped : Nat := 2
```